### PR TITLE
Add ViT architecture

### PR DESCRIPTION
**Pull request by Factral**
### Description
This PR adds a python notebook implementing **Vision Transformers (ViT)** architecture https://arxiv.org/pdf/2010.11929  from scratch using PyTorch. 

### Checklist
Ensure your pull request meets the following requirements:
- [x] I have reviewed existing notebooks to avoid duplications.
- [x] My notebook is fully executed.
- [x] I have included a list of all libraries and their versions used in the notebook.
- [x] The notebook file is named clearly to reflect the topic _(e.g., `cross-entropy-loss.ipynb`)._
- [x] I have placed the notebook in the appropriate directory under `notebooks/`.
- [x] My code is simple, precise, and adheres to good coding practices.

### CODE REVIEW

File: automation/notebooks-table-data.csv
Comment (guillepinto) [inline automation/notebooks-table-data.csv:1]: Please write the name of your notebook in [lowercase](https://github.com/semilleroCV/deep-learning-notes/blob/main/CONTRIBUTING.md#3-make-your-changes).

File: notebooks/architectures/ViT.ipynb
Comment (guillepinto) [inline notebooks/architectures/ViT.ipynb:1]: 1. There are some typing errors in your notebook:

image

2. The `datasets` and `transforms` libraries are not being used, I suggest removing them.

**Review by guillepinto — CHANGES_REQUESTED** [review]

File: automation/notebooks-table-data.csv
Comment (Factral) [reply automation/notebooks-table-data.csv:1]: solved in [1369e86](https://github.com/semilleroCV/deep-learning-notes/pull/20/commits/1369e86b525d3560e838a60044d9f79936a2b22c)

File: notebooks/architectures/ViT.ipynb
Comment (Factral) [reply notebooks/architectures/ViT.ipynb:1]: solved in [1369e86](https://github.com/semilleroCV/deep-learning-notes/pull/20/commits/1369e86b525d3560e838a60044d9f79936a2b22c)

Comment (guillepinto) [inline notebooks/architectures/ViT.ipynb:1]: you also have to change the name of the notebook 😅  (ViT.ipynb -> vit.ipynb)

**Review by guillepinto — CHANGES_REQUESTED** [review]

File: notebooks/architectures/ViT.ipynb
Comment (Factral) [reply notebooks/architectures/ViT.ipynb:1]: I had already made the change in the previous commit but it seems that git does not differentiate between lower and upper case when updating a file in a commit. anyways, solved in [199be0f](https://github.com/semilleroCV/deep-learning-notes/pull/20/commits/199be0f39b7472a395b188c35a6a47d6b97d9095)

**Review by guillepinto — COMMENTED** [review]
thanks, great notebook!
